package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,587 @@
+        self._frame = -1
+        if isinstance(self.filename, h5py.File):
+            self._file = self.filename
+            self._driver = self._file.driver
+        else:
+            if self._comm is not None:
+                # can only pass comm argument to h5py.File if driver='mpio'
+                assert self._driver == 'mpio'
+                self._file = h5py.File(self.filename, 'r',
+                                       driver=self._driver,
+                                       comm=self._comm)
+            elif self._driver is not None:
+                self._file = h5py.File(self.filename, 'r', driver=self._driver)
+            else:
+                self._file = h5py.File(self.filename, 'r')
+        # pulls first key out of 'particles'
+        # allows for arbitrary name of group1 in 'particles'
+        self._particle_group = self._file['particles'][
+                               list(self._file['particles'])[0]]
+
+    def close(self):
+        """close reader"""
+        if self._driver != 'mpio':
+            self._file.close()
+
+    def _reopen(self):
+        """reopen trajectory
+
+        Note
+        ----
+        This method does close the and reopen the file like most other
+        coordinate readers, as this would cause problems with reopening
+        the file with the ``driver`` and ``comm`` arguments for parallel
+        reads. Instead, it rewinds the trajectory back to the first timstep
+        if 'mpio' is passed for the driver.
+
+        """
+        if self._driver != 'mpio':
+            self.close()
+            self.open_trajectory()
+        else:
+            self._frame = -2
+            self._read_next_timestep()
+
+    @property
+    def has_positions(self):
+        """True if 'position' group is in trajectory."""
+        return self._has['position']
+
+    @has_positions.setter
+    def has_positions(self, value: bool):
+        self._has['position'] = value
+
+    @property
+    def has_velocities(self):
+        """True if 'velocity' group is in trajectory."""
+        return self._has['velocity']
+
+    @has_velocities.setter
+    def has_velocities(self, value: bool):
+        self._has['velocity'] = value
+
+    @property
+    def has_forces(self):
+        """True if 'force' group is in trajectory."""
+        return self._has['force']
+
+    @has_forces.setter
+    def has_forces(self, value: bool):
+        self._has['force'] = value
+
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        for name, value in self._has.items():
+            if value:
+                return self._particle_group[name]['value'].shape[0]
+
+    def _read_frame(self, frame):
+        """reads data from h5md file and copies to current timestep"""
+        try:
+            for name, value in self._has.items():
+                if value:
+                    myframe = self._particle_group[name]['step'][frame]
```

Comment:
If we're not using `myframe`, make it clear with
```suggestion
                    _ = self._particle_group[name]['step'][frame]
```
One could also leave out the assignment but I prefer to see that we would be pulling something from the file.

Btw, is this operation here expensive? What type of object is `myframe`?

Reply:
Ahh, I didn't even notice `myframe` wasn't being used... my fault for not catching that. It's an artifact of `GSDReader` that I used as an initial template: https://github.com/MDAnalysis/mdanalysis/blob/2f16a16aa3b9484ba37c9281e32bc80a18e5878e/package/MDAnalysis/coordinates/GSD.py#L99-L110 but it's actually used in `GSDReader` as the object positions and box are pulled from, similar to how I used `self._particle_group`

The dtype is `int32`, and I'm not sure how to tell if it's an expensive operation. Is it still useful to leave the assignment in as a check to see if position/velocity/force exists in the file and that we are able to pull data from the file?